### PR TITLE
Add documentation for hardswish

### PR DESCRIPTION
**Pull request by choidongyeon**
Fix issue https://github.com/pytorch/pytorch/issues/37431.

### CODE REVIEW

**Review by zou3519 — CHANGES_REQUESTED** [review]
Thanks for the PR, @choidongyeon. I don't think adding to the quantization docs is sufficient, because hardswish also works without quantization.

Please add nn.functional.hardswish to https://github.com/pytorch/pytorch/blob/master/docs/source/nn.functional.rst

and nn.Hardswish to https://github.com/pytorch/pytorch/blob/master/docs/source/nn.rst so that it appears in the regular documentation.

**Comment by choidongyeon** [comment]
@zou3519  Hello again :) Thank you for taking a look at this PR! I made the relevant changes.

**Comment by zou3519** [comment]
I think the math in the documentation is wrong (this is a pre-existing problem): 
- https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/activation.py#L333-L338
- https://github.com/pytorch/pytorch/blob/master/torch/nn/functional.py#L1685-L1696

From reading the [hardswish code](https://github.com/pytorch/pytorch/blob/ff9a809ccd5bd61f18dfaa04d41b681e600b3a02/aten/src/ATen/native/cpu/Activation.cpp#L429), it looks like it should be `x(x + 3) / 6`, let me know if you agree. We should correct the documentation before exposing it.

**Comment by choidongyeon** [comment]
> I think the math in the documentation is wrong (this is a pre-existing problem):
> 
> * https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/activation.py#L333-L338
> * https://github.com/pytorch/pytorch/blob/master/torch/nn/functional.py#L1685-L1696
> 
> From reading the [hardswish code](https://github.com/pytorch/pytorch/blob/ff9a809ccd5bd61f18dfaa04d41b681e600b3a02/aten/src/ATen/native/cpu/Activation.cpp#L429), it looks like it should be `x(x + 3) / 6`, let me know if you agree. We should correct the documentation before exposing it.

I agree with you. Just skimmed through [the referenced paper](https://arxiv.org/abs/1905.02244). And based on that,  when `abs(x) < 3`, 

```
HardSwish(x) = x * RELU6(x+3) / 6
= x * min(max(x + 3,0), 6)/ 6
= x * min(x + 3, 6) /6
= x * (x + 3) 6.
```
So it's not just the code. Will change and  update PR.

File: docs/source/nn.functional.rst [hunk 1]
```diff
@@ -160,6 +160,12 @@ Non-linear activation functions
 .. autofunction:: hardtanh
 .. autofunction:: hardtanh_
 
+:hidden:`hardswish`
+~~~~~~~~~~~~~~~~~~
+
+.. autofunction:: hardswish
+.. autofunction:: hardswish_
```
Comment (zou3519) [inline docs/source/nn.functional.rst:167]: I'm getting module 'torch.nn.functional' has no attribute 'hardswish_' when I run "torch.nn.functional.hardswish_". I don't think it is an actual op

**Review by zou3519 — CHANGES_REQUESTED** [review]
One last thing (see comment) and then we should be good to go after.

**Comment by choidongyeon** [comment]
@zou3519 I think we're good to go! Pending two checks that I don't think should be affected by my changes?

**Review by zou3519 — APPROVED** [review]
looks good, thanks!

**Pull request closed without merging**